ui/components/src/LineChart/LineChart.tsx

```diff
@@ -73,6 +73,7 @@ export interface LineChartProps {
   grid?: GridComponentOption;
   legend?: LegendComponentOption;
   tooltipConfig?: TooltipConfig;
+  noDataVariant?: 'chart' | 'message';
```

Comment:
The other option I can think of is a more nested `variant` object, which would allow for customization of other properties in the future (might be overkill / pre-optimization though):
```ts
variant: {
  noData: 'chart' | 'message';
  <other future variants>
}
```

Which makes me second-guess the way I named that `tooltipConfig` prop too 🤔

Reply:
If we did have a variant object like that, it could potentially help with my other comment about the dependency array growing since we could potentially move other props inside it (although not sure which ones currently, we can probably just get rid of ‘legend’ now since no one should be using the ECharts legend anymore)